tests/components/apsystemsapi_local/test_config_flow.py

```diff
@@ -0,0 +1,73 @@
+"""Test the APsystems Local API config flow."""
+
+import aiohttp
+
+from homeassistant import config_entries
+from homeassistant.components.apsystemsapi_local.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_IP_ADDRESS, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+SUCCESS_DEVICE_INFO_DATA = {
+    "data": {
+        "deviceId": "SOME_ID",
+        "devVer": "EZ1 1.6.0",
+        "ssid": "SOME_WIFI",
+        "ipAddr": "SOME_IP",
+        "minPower": "30",
+        "maxPower": "800",
+    },
+    "message": "SUCCESS",
+    "deviceId": "SOME_ID",
+}
+
+
+async def test_form_cannot_connect(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test we handle cannot connect error."""
+    aioclient_mock.get(
+        "http://127.0.0.2:8050/getDeviceInfo", exc=aiohttp.ClientConnectionError
+    )
```

Comment:
You should patch the lib instead of mocking http calls.

Reply:
The unit tests should also be way more complete. One for the happy flow and one for every bad one.

Reply:
But the example wanted me to create a test making sure the flow can recover from an error and it just wouldn't make sense copying the failing part.